src/main/java/seedu/weme/logic/LogicManager.java

```diff
@@ -108,7 +108,9 @@ public LikeData getLikeData() {
         return model.getObservableLikeData();
     }
 
+    @Override
     public void cleanUp() {
-        model.cleanMemeStorage();
+        new Thread(() -> model.cleanMemeStorage()).start();
+        model.cleanTemplateStorage();
```

Comment:
Should this be a in thread too?

Reply:
hmm was thinking that after this thread creates a new thread to clean meme storage it should do some of the work itself. 

Would it make more sense for this thread to just summon the new threads to clean up the resources then die off? It does look less like a bug that way, but it's one extra thread spawned

Reply:
I don't really understand threads. Does the GUI hang if this method does the cleaning itself? My intention was to prevent the GUI from hanging.

Reply:
Nope, the GUI calls a thread to run the cleanUp method. 

The thread that runs the cleanUp method, creates these threads to clean up the storage